### PR TITLE
Update release-process.md

### PR DESCRIPTION
**Pull request by oblm**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
Updates the release process to reflect reality.
* Removes ThirdPersonShooter, which is being deprecated.
* Moves workflow tests that were originally run in ThirdPersonShooter to Stater Project.
* Acknowledges the existence of `preview`.
* Simplifies the `git` workflow.

### CODE REVIEW

**Comment by oblm** [comment]
someone ptal & this, it's getting stale

File: SpatialGDK/Extras/internal-documentation/release-process.md [hunk 1]
```diff
@@ -29,9 +29,15 @@ This document outlines the process for releasing a version of the GDK for Unreal
 1. `git pull`
 1. Using `git log`, take note of the latest commit hash.
 1. `git checkout -b x.y.z-rc` in order to create release candidate branch.
-1. Open `CHANGELOG.md`, which is in the root of the repository, and put the release version and planned date of release in a `##` block. Move the `Unreleased` section above this.
+1. Open `CHANGELOG.md`, which is in the root of the repository.
+1. Read **every** release note in the `Unreleased` section. Ensure that they make sense, they conform to [how-to-write-good-release-notes.md ](https://github.com/spatialos/UnrealGDK/blob/master/SpatialGDK/Extras/internal-documentation/how-to-write-good-release-notes.md) structure.
```
Comment (m-samiec) [inline SpatialGDK/Extras/internal-documentation/release-process.md]:
```suggestion
1. Read **every** release note in the `Unreleased` section. Ensure that they make sense, they conform to [how-to-write-good-release-notes.md](https://github.com/spatialos/UnrealGDK/blob/master/SpatialGDK/Extras/internal-documentation/how-to-write-good-release-notes.md) structure.
```

File: SpatialGDK/Extras/internal-documentation/release-process.md [hunk 2]
```diff
@@ -144,27 +130,44 @@ The workflow for this is:
...
-If you want to soak test this release in `staging-` before releasing to `master`, only execute the `staging-` steps.
+If you want to soak test this release on the `preview` branch before promoting it to the `release` branch, only execute the steps that merge into `preview` and `master`.
 
 When merging the following PRs, you need to enable `Allow merge commits` option on the repos and choose `Create a merge commit` from the dropdown in the pull request UI to merge the branch, then disable `Allow merge commits` option on the repos once the release process is complete. You need to be an admin to perform this.
 
-1. In `UnrealGDK`, create `staging-x.y.z-rc` from `x.y.z-rc` (select `x.y.z-rc` in the branch dropdown, then type the name of the new branch to create. If there's already a branch with that name, delete it first).
-1. In `UnrealGDK`, merge `x.y.z-rc` into `release` using GitHub PR (use `Update branch` button to merge `release` into `x.y.z-rc`).
-1. Use the GitHub Release UI to tag the commit you just made to as `x.y.z`.<br/>
+**UnrealGDK**
+1. In `UnrealGDK`, merge `x.y.z-rc` into `master`.
```
Comment (m-samiec) [inline SpatialGDK/Extras/internal-documentation/release-process.md]: I don't believe this is correct. We merge first into release, and then merge release into master. @improbable-valentyn to confirm.

File: SpatialGDK/Extras/internal-documentation/release-process.md [hunk 3]
```diff
@@ -144,27 +130,44 @@ The workflow for this is:
...
+1. In `UnrealGDK`, merge `x.y.z-rc` into `preview`.
+1. Use the [GitHub Release UI](https://github.com/spatialos/UnrealGDK/releases) to tag the commit you just made to `preview` as `x.y.z-preview`.<br/>
+Copy the latest release notes from `CHANGELOG.md` and paste them into the release description field.
+1. In `UnrealGDK`, merge `x.y.z-rc` into `release`.
+1. Use the [GitHub Release UI](https://github.com/spatialos/UnrealGDK/releases) to tag the commit you just made to `release` as `x.y.z`.<br/>
+Copy the latest release notes from `CHANGELOG.md` and paste them into the release description field.
+
+**improbableio/UnrealEngine**
+1. In `improbableio/UnrealEngine`, merge `4.xx-SpatialOSUnrealGDK-x.y.z-rc` into `4.xx-SpatialOSUnrealGDK`.
```
Comment (m-samiec) [inline SpatialGDK/Extras/internal-documentation/release-process.md]: Likewise here

File: SpatialGDK/Extras/internal-documentation/release-process.md (hunk 2 above)
Comment (oblm) [reply SpatialGDK/Extras/internal-documentation/release-process.md]: good spot, @improbable-valentyn ptal

Comment (improbable-valentyn) [reply SpatialGDK/Extras/internal-documentation/release-process.md]: Mike's right, we should merge back into master after the release. Note that especially for the `UnrealGDK` repo, this merge could have conflicts and so will require an engineer to resolve.

Comment (oblm) [reply SpatialGDK/Extras/internal-documentation/release-process.md]: fixed

File: SpatialGDK/Extras/internal-documentation/release-process.md (hunk 3 above)
Comment (oblm) [reply SpatialGDK/Extras/internal-documentation/release-process.md]: fixed, also fixed in the Example Project section

**Comment by oblm** [comment]
Chatted with @improbable-valentyn who raised the following questions:

• Do we have the concept of "release stages"? i.e, is a commit identical branch likely to be promoted from `preview` to `release`?
• If a full release is tagged on the `release` branch, should it also be tagged on the `preview` branch although it was never released in a preview state?
• Depending on our answer to the above, is `preview` still an appropriate name for this branch.

I'll chat about this with the team, build consensus and make any necessary changes prior to merging.

**Comment by oblm** [comment]
> Chatted with @improbable-valentyn who raised the following questions:
> 
> • Do we have the concept of "release stages"? i.e, is a commit identical branch likely to be promoted from `preview` to `release`?
> • If a full release is tagged on the `release` branch, should it also be tagged on the `preview` branch although it was never released in a preview state?
> • Depending on our answer to the above, is `preview` still an appropriate name for this branch.
> 
> I'll chat about this with the team, build consensus and make any necessary changes prior to merging.

these decisions have now been made and implemented in this PR, this is ready to merge.

File: SpatialGDK/Extras/internal-documentation/release-process.md [hunk 4]
```diff
@@ -144,27 +136,48 @@ The workflow for this is:
...
 
-When merging the following PRs, you need to enable `Allow merge commits` option on the repos and choose `Create a merge commit` from the dropdown in the pull request UI to merge the branch, then disable `Allow merge commits` option on the repos once the release process is complete. You need to be an admin to perform this.
+1. When merging the following PRs, you need to enable `Allow merge commits` option on the repos and choose `Create a merge commit` from the dropdown in the pull request UI to merge the branch, then disable `Allow merge commits` option on the repos once the release process is complete. You need to be an admin to perform this.
+
+**UnrealGDK**
+1. In `UnrealGDK`, merge `x.y.z-rc` into `preview`.
+1. If you want to soak test this release on the `preview`, use the [GitHub Release UI](https://github.com/spatialos/UnrealGDK/releases) to tag the commit you just made to `preview` as `x.y.z-preview`.<br/>
+Copy the latest release notes from `CHANGELOG.md` and paste them into the release description field.
+1. In `UnrealGDK`, merge `x.y.z-rc` into `release`.
```
Comment (improbable-valentyn) [inline SpatialGDK/Extras/internal-documentation/release-process.md]:
```suggestion
1. In `UnrealGDK`, merge `preview` into `release`.
```

File: SpatialGDK/Extras/internal-documentation/release-process.md [hunk 5]
```diff
@@ -144,27 +136,48 @@ The workflow for this is:
...
+1. Use the [GitHub Release UI](https://github.com/spatialos/UnrealGDK/releases) to tag the commit you just made to `release` as `x.y.z`.<br/>
+Copy the latest release notes from `CHANGELOG.md` and paste them into the release description field.
+1. In `UnrealGDK`, merge `release` into `master`. This merge could have conflicts. Don't hesitate to ask for help resolving these if you are unsure.
+
+**improbableio/UnrealEngine**
+1. In `improbableio/UnrealEngine`, merge `4.xx-SpatialOSUnrealGDK-x.y.z-rc` into `4.xx-SpatialOSUnrealGDK-preview`.
+1. If you want to soak test this release on the `preview`, use the [GitHub Release UI](https://github.com/improbableio/UnrealEngine/releases) to tag the commit you just made to `4.xx-SpatialOSUnrealGDK-preview` as `4.xx-SpatialOSUnrealGDK-x.y.z-preview`.<br/>
+Copy the latest release notes from `CHANGELOG.md` and paste them into the release description field.
+1. In `improbableio/UnrealEngine`, merge `4.xx-SpatialOSUnrealGDK-x.y.z-rc` into `4.xx-SpatialOSUnrealGDK-release`.
```
Comment (improbable-valentyn) [inline SpatialGDK/Extras/internal-documentation/release-process.md]:
```suggestion
1. In `improbableio/UnrealEngine`, merge `4.xx-SpatialOSUnrealGDK-preview` into `4.xx-SpatialOSUnrealGDK-release`.
```

File: SpatialGDK/Extras/internal-documentation/release-process.md [hunk 6]
```diff
@@ -144,27 +136,48 @@ The workflow for this is:
...
+1. In `improbableio/UnrealEngine`, merge `4.xx-SpatialOSUnrealGDK-release` into `master`. This merge could have conflicts. Don't hesitate to ask for help resolving these if you are unsure.
 
-1. In `UnrealGDK`, create `staging-x.y.z-rc` from `x.y.z-rc` (select `x.y.z-rc` in the branch dropdown, then type the name of the new branch to create. If there's already a branch with that name, delete it first).
-1. In `UnrealGDK`, merge `x.y.z-rc` into `release` using GitHub PR (use `Update branch` button to merge `release` into `x.y.z-rc`).
-1. Use the GitHub Release UI to tag the commit you just made to as `x.y.z`.<br/>
+**UnrealGDKExampleProject**
+1. In `UnrealGDKExampleProject`, merge `x.y.z-rc` into `preview`.
+1. If you want to soak test this release on the `preview`, use the [GitHub Release UI](https://github.com/spatialos/UnrealGDKExampleProject/releases) to tag the commit you just made to `preview` as `x.y.z-preview`.<br/>
+Copy the latest release notes from `CHANGELOG.md` and paste them into the release description field.
+1. In `UnrealGDKExampleProject`, merge `x.y.z-rc` into `release`.
```
Comment (improbable-valentyn) [inline SpatialGDK/Extras/internal-documentation/release-process.md]:
```suggestion
1. In `UnrealGDKExampleProject`, merge `preview` into `release`.
```

**Pull request merged**